### ISSUE TITLE
minifier: case variable body is incorrectly inlined to other cases

### ISSUE THREAD

**Issue by armano2** [oxc-project/oxc#26174]
```js
switch (a) { case 2: let x = 1; case 1: console.log(x); }
switch (a) { case 2: case 1: console.log(1); }
```

when discriminant is 1, it should throw error and when its 2 its should pass

related to #24659 but for switches
